eureka-core/src/main/java/com/netflix/eureka/util/EurekaMonitors.java

```diff
@@ -174,6 +192,26 @@ public static void registerAllStats() {
         }
     }
 
+    /**
+     * Registers metrics to given Spectator registry
+     * @param registry Spectator registry to register counters to
+     */
+
+    public static void registerAllStatsSpectator(Registry registry) {
```

Comment:
It looks like these are just here to allow access to the registry from a static context. Spectator has a notion of a [global registry](http://netflix.github.io/spectator/en/latest/intro/registry/#global-registry) for that purpose. This would have two main benefits:

1. When configuring the registry the user doesn't need to now about these specific methods scattered around various classes.
2. You could get rid of the null checks for the use. It will default to a noop implementation if no registries are added.